language/move-lang/src/interface_generator.rs

```diff
@@ -247,7 +247,28 @@ fn write_ability(ab: Ability) -> String {
     .to_string()
 }
 
-fn write_type_parameters(tps: &[AbilitySet]) -> String {
+fn write_struct_type_parameters(tps: &[TypeParameter]) -> String {
+    if tps.is_empty() {
+        return "".to_string();
+    }
+
+    let tp_and_constraints = tps
+        .iter()
+        .enumerate()
+        .map(|(idx, ty_param)| {
+            format!(
+                "{}{}{}",
+                if ty_param.is_phantom { "phantom " } else { "" },
```

Comment:
this is probably fine. but would also be fine with doing an `assert!(ty_param.is_phantom())` or something for now. Up to you though since you will likely be doing both parts :)